core/lib/external_price_api/src/utils.rs

```diff
@@ -13,3 +13,33 @@ pub fn get_fraction(ratio_f64: f64) -> (NonZeroU64, NonZeroU64) {
 
     (numerator, denominator)
 }
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::*;
+
+    fn assert_get_fraction_value(f: f64, num: u64, denum: u64) {
+        assert_eq!(
+            get_fraction(f),
+            (
+                NonZeroU64::try_from(num).unwrap(),
+                NonZeroU64::try_from(denum).unwrap()
+            )
+        );
+    }
+
+    #[allow(clippy::approx_constant)]
+    #[test]
+    fn test_float_to_fraction_conversion_as_expected() {
+        assert_get_fraction_value(1.0, 1, 1);
+        assert_get_fraction_value(1337.0, 1337, 1);
+        assert_get_fraction_value(0.1, 1, 10);
+        assert_get_fraction_value(3.141, 3141, 1000);
+        assert_get_fraction_value(1_000_000.0, 1_000_000, 1);
+        assert_get_fraction_value(3123.47, 312347, 100);
+        // below tests assume some not necessarily required behaviour of get_fraction
+        assert_get_fraction_value(0.2, 1, 5);
+        assert_get_fraction_value(0.5, 1, 2);
+        assert_get_fraction_value(3.1415, 6283, 2000);
+    }
```

Comment:
Should include some `#[should_panic]` tests for zero num/den

Reply:
We should never get panic from get_fraction as it converts from float to fraction so there should never be 0 denum, but I should test the value of zero and negative values also for completeness. Good catch!

Reply:
Actually we are panicing sometimes! I believe we might not want to.